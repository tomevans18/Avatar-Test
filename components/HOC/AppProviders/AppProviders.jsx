import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import theme from '../../../styles/theme';

import ResizeAnimationHandler from '../ResizeAnimationHandler';

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
};

const AppProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <ResizeAnimationHandler>{children}</ResizeAnimationHandler>
  </ThemeProvider>
);

AppProviders.propTypes = propTypes;

export default AppProviders;
