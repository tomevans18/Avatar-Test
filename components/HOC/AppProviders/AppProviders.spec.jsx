import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AppProviders from './AppProviders';

let MockComponent;

describe('App Provider', () => {
  beforeEach(() => {
    MockComponent = () => <div />;
  });

  it('should render as expected', () => {
    const wrapper = shallow(
      <AppProviders>
        <MockComponent />
      </AppProviders>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
