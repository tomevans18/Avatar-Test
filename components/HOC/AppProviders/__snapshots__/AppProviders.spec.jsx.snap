// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Provider should render as expected 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoint": Object {
        "extraLarge": 1224,
        "large": 1025,
        "medium": 769,
        "small": 451,
      },
      "color": Object {
        "available": "rgb(46, 204, 113)",
        "background": "rgb(53, 59, 72)",
        "black": "#1e1e1e",
        "blue": "#2d8bd8",
        "grey": "rgb(220, 221, 225)",
        "grey_light": "#eaeaea",
        "notReady": "rgb(251, 197, 49)",
        "unavailable": "rgb(194, 54, 22)",
        "white": "#ffffff",
      },
      "component": Object {
        "spacing": "1.25em",
      },
    }
  }
>
  <UserProvider>
    <ResizeAnimationHandler>
      <MockComponent />
    </ResizeAnimationHandler>
  </UserProvider>
</ThemeProvider>
`;
